shaders/src/depth_main.fs

```diff
@@ -17,16 +17,16 @@ void main() {
 #if defined(HAS_VSM)
     // Since we're rendering from the perspective of the light, frameUniforms.cameraPosition is the
     // light position, in world space.
-    // We need a linear depth representation, so we can't simply use gl_FragDepth here, which won't
-    // be linear for spot shadows or when using LiSPM.
+    // We use "distance to the light" as the depth metric, which works for both directional and spot
+    // lights.
     highp float depth = length(frameUniforms.cameraPosition.xyz - vertex_worldPosition);
 
     highp float dx = dFdx(depth);
     highp float dy = dFdy(depth);
 
     // Output the first and second depth moments.
     // The first moment is mean depth.
-    // The second moment is depth squared.
+    // The second moment is mean depth squared.
     // These values are retrieved when sampling the shadow map to compute variance.
     highp float bias = 0.25 * (dx * dx + dy * dy);
```

Comment:
is the 0.25 the correct value, or is it a settable?

Reply:
It's a constant value. Taken from section 8.4.2: https://developer.nvidia.com/gpugems/gpugems3/part-ii-light-and-shadows/chapter-8-summed-area-variance-shadow-maps